### ISSUE TITLE
FreezedUnionValue accepts only String

### ISSUE THREAD

**Issue by paraddise** [rrousselGit/freezed#564]
**Is your feature request related to a problem? Please describe.**
In my firestore key`type` is number, and I want it to be `unionKey`, but `FreezedUnionValue` accepts only `String`.

**Describe the solution you'd like**
```dart
class FreezedUnionValue<T> {
  const FreezedUnionValue(this.value);

  final T value;
}
```

**Describe alternatives you've considered**
Alternatively I can store `type` as string, but i think that bad idea

**Comment by onism0106** [comment]
Yes, I have this requirement as well. Sometimes I need to set the FreezedUnionValue to an int value.
